drivers/sensor/nordic/qdec_nrfx/qdec_nrfx.c

```diff
@@ -173,81 +174,78 @@ static const struct sensor_driver_api qdec_nrfx_driver_api = {
 	.trigger_set  = qdec_nrfx_trigger_set,
 };
 
-#ifdef CONFIG_PM_DEVICE
-static int qdec_nrfx_pm_action(const struct device *dev,
-			       enum pm_device_action action)
+static int qdec_pm_suspend(const struct device *dev)
 {
 	const struct qdec_nrfx_config *config = dev->config;
-	int ret = 0;
+
+	nrfx_qdec_disable(&config->qdec);
+	qdec_nrfx_gpio_ctrl(dev, false);
+
+	if (!IS_ENABLED(CONFIG_PM_DEVICE)) {
+		return 0;
+	}
+
+	return pinctrl_apply_state(config->pcfg, PINCTRL_STATE_SLEEP);
```

Comment:
note: this requires a `sleep` state to be defined, if not, device will fail to suspend (due to -ENOENT). This will happen on nrf54l board according to next commit. Since pinctrl if not fallible in nrf now, we may just ignore return code and relax pin state requirements.

Reply:
reworked to ignore return value.